packages/lambdas/src/fhir-to-medical-record.ts

```diff
@@ -96,209 +97,1282 @@ export const handler = Sentry.AWSLambda.wrapHandler(
+                ${createHeaderTableRow("DOB", patient.birthDate ?? "")}
+              </tbody>
+            </table>
+          </div>
+          <div>
+            <h4>Author</h4>
+            <table class="header-table-author">
+              <tbody>
+                ${createHeaderTableRow("Name", "Metriport")}
+                ${createHeaderTableRow("Authored On", dayjs().format("YYYY-MM-DD"))}
+              </tbody>
+            </table>
+          </div>
+        </div>
+        <div class="header-table">
+          <h4>Table of Contents</h4>
+          <ul id="nav">
+            <div class='half'>
+              <li>
+                <a href="#medications">Medications</a>
+              </li>
+              <li>
+                <a href="#conditions">Conditions</a>
+              </li>
+              <li>
+                <a href="#allergies">Allergies</a>
+              </li>
+              <li>
+                <a href="#procedures"
+                  >Procedures</a
+                >
+              </li>
+              <li>
+                <a href="#social-history">Social History</a>
+              </li>
+              <li>
+                <a href="#vitals">Vitals</a>
+              </li>
+              <li>
+                <a href="#laboratory">Laboratory</a>
+              </li>
+            </div>
+            <div class='half'>
+              <li>
+                <a href="#encounters">Encounters</a>
+              </li>
+              <li>
+                <a href="#immunizations">Immunizations</a>
+              </li>
+              <li>
+                <a href="#family-member-history">Family Member History</a>
+              </li>
+              <li>
+                <a href="#related-persons">Related Persons</a>
+              </li>
+              <li>
+                <a href="#tasks">Tasks</a>
+              </li>
+              <li>
+                <a href="#coverage">Coverage</a>
+              </li>
+            </div>
+            </ul>
+        </div>
+      </div>
+    </div>
+  `;
 }
 
-async function getBundleFromS3(fileName: string) {
-  const getResponse = await s3Client
-    .getObject({
-      Bucket: bucketName,
-      Key: fileName,
+function createHeaderTableRow(label: string, value: string) {
+  return `
+    <tr>
+      <td>
+        <span class="header-label">${label}</span>
+        <span>
+          ${value}
+        </span>
+      </td>
+    </tr>
+  `;
+}
+
+function createMedicationSection(fhirBundle: Bundle) {
+  const medicationRequests: MedicationRequest[] | undefined = fhirBundle.entry
+    ?.filter(entry => {
+      return entry?.resource?.resourceType === "MedicationRequest";
     })
-    .promise();
-  const objectBody = getResponse.Body;
-  if (!objectBody) throw new Error(`No body found for ${fileName}`);
-  return JSON.parse(objectBody.toString());
+    .map(entry => {
+      return entry.resource as MedicationRequest;
+    });
```

Comment:
ditto, see comment above, this looping/filtering/mapping can be reduced to a single loop at the top of the function - so, we won't have to do it for every section